### PR TITLE
Remove overridden DMLC methods accepting Queue

### PR DESCRIPTION
**Pull request by panchenko**
As the implementations of those in AbstractMessageListenerContainer
delegate to ones accepting Strings.

### CODE REVIEW

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java [hunk 1]
```diff
@@ -245,21 +244,20 @@ public AcknowledgeMode getAcknowledgeMode() {
...
+		this.queueNames = new CopyOnWriteArrayList<>(queueName);
 	}
 
 	/**
 	 * Set the name of the queue(s) to receive messages from.
 	 * @param queues the desired queue(s) (can not be <code>null</code>)
 	 */
 	public void setQueues(Queue... queues) {
-		List<String> queueNames = new ArrayList<String>(queues.length);
+		final String[] queueNames = new String[queues.length];
```
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java]: Can't we use `Arrays.stream(queues).map(Queue::getName)` instead ?

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java [hunk 2]
```diff
@@ -196,12 +196,6 @@ public final void setMissingQueuesFatal(boolean missingQueuesFatal) {
 	}
 
 	@Override
-	public void setQueues(Queue... queues) {
-		Assert.state(!isRunning(), "Cannot set queue names while running, use add/remove");
```
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java]: ??? Why do you remove this method if it has this extra assertion logic ?

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java [hunk 3]
```diff
@@ -212,18 +206,6 @@ public void addQueueNames(String... queueNames) {
 		super.addQueueNames(queueNames);
 	}
 
-	@Override
-	public void addQueues(Queue... queues) {
-		try {
-			addQueues(Arrays.stream(queues)
```
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java]: ??? This is delegation to the `private void addQueues(Stream<String> queueNameStream)`.
What is the point to get rid of this logic?
Thanks

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java [hunk 4]
```diff
@@ -248,19 +230,12 @@ public boolean removeQueueNames(String... queueNames) {
 		return super.removeQueueNames(queueNames);
 	}
 
-	@Override
-	public boolean removeQueues(Queue... queues) {
-		removeQueues(Arrays.stream(queues)
```
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java]: The same here. To the `private void removeQueues(Stream<String> queueNames) `

What am I missing, please ?

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java (hunk 2 above)
Comment (panchenko) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java]: Currently both `setQueues(Queue...)` and `setQueueNames(String...)` are overridden with the same checks.

However methods in the base class collect names from `Queue...` and call the corresponding method with `String...` - I tried to mentioned that above, sorry it it was not clear.
So, it's enough to have this extra logic only in the methods with `String...`.

File: .travis.yml [hunk 5]
```diff
@@ -4,6 +4,8 @@ sudo: false
 services:
   - rabbitmq
 env:
+  global:
+  - GRADLE_OPTS="-Xmx512m"
```
Comment (artembilan) [inline .travis.yml]: What is the purpose of this change?

Would be great to have itemization in the commit message with the reason of each modification.
At least those which are not obvious.
Thanks

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java (hunk 3 above)
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java]: You haven't answered to this concerns yet.
Thanks

File: .travis.yml (hunk 5 above)
Comment (panchenko) [inline .travis.yml]: It failed once in Travis with some "killed" message, my guess is that main gradle process used too much memory.

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java (hunk 3 above)
Comment (panchenko) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java]: There are 3 methods - set/add/remove, and those with `Queue...` delegate to ones with `String...`, which actually do all the work.
IMHO methods with `Queue...` can even be declared final.

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java (hunk 4 above)
Comment (panchenko) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java]: This one also delegates to `removeQueueNames(String...)` which has this extra logic.

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java (hunk 3 above)
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java]: OK! After closer look I really see your logic and what you suggest.
Not enough coffee? Or to many snow outside 😄 

Yes, I think we can go ahead with your suggestion and even really make those `Queue`-based methods as `final`.

Thank you for such a comprehensive analysis and patience to my nit-pickings.

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java [hunk 6]
```diff
@@ -293,13 +296,7 @@ public void addQueueNames(String... queueNames) {
...
 	 */
 	public void addQueues(Queue... queues) {
-		Assert.notNull(queues, "'queues' cannot be null");
-		Assert.noNullElements(queues, "'queues' cannot contain null elements");
-		String[] queueNames = new String[queues.length];
-		for (int i = 0; i < queues.length; i++) {
-			queueNames[i] = queues[i].getName();
-		}
-		this.addQueueNames(queueNames);
+		this.addQueueNames(collectQueueNames(queues));
```
Comment (garyrussell) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java]: We don't use `this.` on method calls; previous code was bad; we generally clean these up when modifying code.

Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java]: We can do that on merge.
But since you, @panchenko , are so active contributor, please, consider to follow our code style and overall tracking rules to avoid similar comments again and again.
That should save some time for us, as well as for you 😄

Comment (panchenko) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java]: Sure I can adjust this :-)

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-amqp/commit/eeab98e3657f675126718fdebd896e90b8960872
